inc/Lazyload/Component.php

```diff
@@ -130,20 +137,62 @@ public function action_add_lazyload_filters() {
 	}
 
 	/**
-	 * Enqueues and defer lazy-loading JavaScript.
+	 * Registers and defers lazy-loading JavaScript.
+	 *
+	 * Although the script isn't directly enqueued via WordPress, it should still be registered so that it can be
+	 * precached by the service worker.
 	 */
-	public function action_enqueue_lazyload_assets() {
-		wp_enqueue_script(
+	public function action_register_lazyload_assets() {
+		// phpcs:ignore WordPress.WP.EnqueuedResourceParameters
+		wp_register_script(
 			'wp-rig-lazy-load-images',
 			get_theme_file_uri( '/assets/js/lazyload.min.js' ),
 			[],
-			wp_rig()->get_asset_version( get_theme_file_path( '/assets/js/lazyload.min.js' ) ),
-			false
+			null,
+			true
 		);
 		wp_script_add_data( 'wp-rig-lazy-load-images', 'defer', true );
 		wp_script_add_data( 'wp-rig-lazy-load-images', 'precache', true );
 	}
 
+	/**
+	 * Prints the lazy-loading script.
+	 *
+	 * If the browser supports native lazy-loading via the 'loading' attribute, values of data attributes on the images
+	 * will be written back into their real attributes. If the browser does not support native lazy-loading, the script
+	 * bundled in the theme will be loaded.
+	 *
+	 * @see https://web.dev/native-lazy-loading
+	 */
+	public function action_print_lazyload_script() {
+		?>
+		<script type="text/javascript">
+			if ( 'loading' in HTMLImageElement.prototype ) {
+				( function() {
+					var images = [].slice.call( document.querySelectorAll( 'img.lazy' ) );
+					images.forEach( function( img ) {
+						img.src = img.dataset.src;
```

Comment:
I think this should potentially be moved to the bottom of the loop. Just thinking that some engines could potentially start requesting the `src` before the `srcset` is set, thus inadvertently downloading two images instead of one.